src/pocketmine/block/WallBanner.php

```diff
@@ -0,0 +1,53 @@
+<?php
+
+/*
+ *
+ *  ____            _        _   __  __ _                  __  __ ____
+ * |  _ \ ___   ___| | _____| |_|  \/  (_)_ __   ___      |  \/  |  _ \
+ * | |_) / _ \ / __| |/ / _ \ __| |\/| | | '_ \ / _ \_____| |\/| | |_) |
+ * |  __/ (_) | (__|   <  __/ |_| |  | | | | | |  __/_____| |  | |  __/
+ * |_|   \___/ \___|_|\_\___|\__|_|  |_|_|_| |_|\___|     |_|  |_|_|
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * @author PocketMine Team
+ * @link http://www.pocketmine.net/
+ *
+ *
+*/
+
+declare(strict_types=1);
+
+namespace pocketmine\block;
+
+use pocketmine\level\Level;
+
+class WallBanner extends StandingBanner{
+
+	protected $id = self::WALL_BANNER;
+
+	public function getName() : string{
+		return "Wall Banner";
+	}
+
+	public function onUpdate(int $type){
+		$faces = [
+			2 => 3,
```

Comment:
Could probably use Vector3 constants here

Reply:
The PillarRotationHelper only works for pillars; It takes 3 rotations in account, facing north, facing east or facing upwards. That doesn't really work with banners that have 4 rotations; north, east, south and west.

Reply:
what about vector3 constants? https://github.com/pmmp/PocketMine-MP/blob/master/src/pocketmine/math/Vector3.php#L30L33

Reply:
For the sake of consistency I think I'm just going to keep it like this; The WallSign class has the exact same piece of code.

Reply:
They probably won't be there for long 😛 
https://github.com/pmmp/PocketMine-MP/commit/c394aea80325d9cf06aab4cee302e4be5a82e56b